### DISCUSSION TITLE
Project Update Q3 2024

### DISCUSSION THREAD

**Discussion by DeeEmm** [DIY-Flow-Bench/DIY-Flow-Bench#170] [Announcements]
Well It's finally done. I've updated the PCB design and ordered some more PCBs. Yay! There has been some recent action on the project, mostly over on the Discord channel with a few additional users building benches which had given me a bit of a kick to get back into the project and work on those long overdue, promised updates.

The PCB update includes fixes for the issues found on the previous board and includes additional footprints for the XGZ style pressure sensors as well as alternate BME footprints. This should hopefully cater for whichever sensors are available for your build.

There is also a code update that I have published to the master branch. This includes a minor fix for the baro display units in the GUI. The update has also spurned me on to do a bit of a code tidy. I'm also taking advantage of artificial intelligence to do some sanity checking on my calculations. 

Using AI as a coding tool is kind of interesting. It wasn't available back when I started this, which meant that I needed to figure it all out on my own. Whilst I do have a background in engineering, fluid dynamics is not my specialist subject. So having the ability to utilise a coding tool like this for free is a very useful thing. Much like being able to peer review my work.

I've update the files in the PCB folder and also the Wiki. Although the Wiki probably needs a more in depth update. The Document folder also now includes a pretty awesome interactive BOM, where you can click on the BOM list and it will show you where the component is on the PCB. A super useful tool when ti comes to assembling the PCB and you want to know what resistor goes where. Check it out...

[https://htmlpreview.github.io/?https://github.com/DeeEmm/DIY-Flow-Bench/blob/master/docs/iBom-DIY-FLOW-BENCH-V2-3.html](https://htmlpreview.github.io/?https://github.com/DeeEmm/DIY-Flow-Bench/blob/master/docs/iBom-DIY-FLOW-BENCH-V2-3.html
)

I'm having an initial batch of 20 PCBs made up and have decided that I will make a small charge of AU$10 + shipping for the PCBs to cover costs.

So that's where things currently stand. I'll continue with the code review and let everyone know when the new PCBs are available.

/DM

**Comment by dobrowolskibartosz** [comment]
Any chance to order the board already or will you publish gerbers?

**Reply by DeeEmm** [reply]
I have received the boards, but not yet validated them. I will let the group know when the they are available to purchase.

At this stage (beta) gerbers and hardware design files will not be published. Schematics are already available should you want to design your own PCB.

**Reply by dobrowolskibartosz** [reply]
OK, i will wait for you to validate the boards. Interested in 2 :)

**Comment by DeeEmm** [comment]
Just thought I'd update this post with what has been going on in the project. There has been a number of code updates addressing various bugs and features since RC7 was published. However at present I am holding off from pushing out the RC8 release until issues #178 and #179 are resolved. The flow and leak calibration functions are integral to the correct operation of the bench, so until these have a satisfactory resolution the current code does not meet Release Candidate requirements.

We have had a lot of input from community members which has provided great assistance to general development and a great amount of work has been carried out recently. A big thanks to all of those who have contributed over the past few months. Your help and input is massively appreciated.

I am hopeful that we can provide quick resolutions to issues #178 and #179 and move closer towards the official V2 release.

We still need help with validation testing, so if you have a bench that is running the DIYFB code, please step up and provide feedback on your experience and ask how to undertake the validation tests.

/DM
